vision/cloud-client/src/main/java/com/example/vision/Detect.java

```diff
@@ -0,0 +1,405 @@
+
+    for (AnnotateImageResponse res : responses) {      
+      if (res.hasError()) out.printf("Error: %s\n", res.getError().getMessage());
+
+      for (EntityAnnotation annotation : res.getLandmarkAnnotationsList()) {
+        LocationInfo info = annotation.getLocationsList().listIterator().next();
+        out.printf("Landmark: %s\n %s\n", annotation.getDescription(), info.getLatLng());
+      }
+    }
+  }
+  
+  /**
+   * Detects logos in the specified image.
+   * @param filePath The path to the file to perform logo detection on.
+   * @param out A {@link PrintStream} to write detected logos to.
+   * @throws IOException 
+   */
+  static public void detectLogos(String filePath, PrintStream out) throws IOException {
+    List<AnnotateImageRequest> requests = new ArrayList<>();
+
+    FileInputStream file = new FileInputStream(filePath);
+    ByteString imgBytes = ByteString.copyFrom(toByteArray(file));
+    
+    Image img = Image.newBuilder().setContent(imgBytes).build();    
+    Feature feat = Feature.newBuilder().setType(Type.LOGO_DETECTION).build();
+    AnnotateImageRequest request = AnnotateImageRequest.newBuilder()
+        .addFeatures(feat)
+        .setImage(img)
+        .build();
+    requests.add(request);
+
+    BatchAnnotateImagesResponse response = visionApi.batchAnnotateImages(requests);
+    List<AnnotateImageResponse> responses = response.getResponsesList();
+
+    for (AnnotateImageResponse res : responses) {      
+      if (res.hasError()) out.printf("Error: %s\n", res.getError().getMessage());
+
+      for (EntityAnnotation annotation : res.getLogoAnnotationsList()) {
+        out.println(annotation.getDescription());
+      }
+    }
+  }
+  
+  /**
+   * Detects text in the specified image.
+   * @param filePath The path to the file to detect text in.
+   * @param out A {@link PrintStream} to write the detected text to.
+   * @throws IOException 
+   */
+  static public void detectText(String filePath, PrintStream out) throws IOException {
+    List<AnnotateImageRequest> requests = new ArrayList<>();
+
+    FileInputStream file = new FileInputStream(filePath);
+    ByteString imgBytes = ByteString.copyFrom(toByteArray(file));
+    
+    Image img = Image.newBuilder().setContent(imgBytes).build();    
+    Feature feat = Feature.newBuilder().setType(Type.TEXT_DETECTION).build();
+    AnnotateImageRequest request = AnnotateImageRequest.newBuilder()
+        .addFeatures(feat)
+        .setImage(img)
+        .build();
+    requests.add(request);
+
+    BatchAnnotateImagesResponse response = visionApi.batchAnnotateImages(requests);
+    List<AnnotateImageResponse> responses = response.getResponsesList();
+
+    for (AnnotateImageResponse res : responses) {
+      if (res.hasError()) out.printf("Error: %s\n", res.getError().getMessage());
+
+      for (EntityAnnotation annotation : res.getTextAnnotationsList()) {
+        out.printf("Text: %s\n", annotation.getDescription());
```

Comment:
I believe the api returns coordinates as well - might be useful to highlight that (eg `"Text ({x}, {y}): %s\n"`), since it's a fairly useful feature and not necessarily obvious that it's there.